### PR TITLE
Allow multiple ml2 type drivers at the same time

### PR DESCRIPTION
**Pull request by toabctl**
This is related to https://github.com/crowbar/barclamp-neutron/pull/175

### CODE REVIEW

**Comment by tboerger** [comment]
LGTM

File: crowbar_framework/app/models/nova_service.rb [hunk 1]
```diff
@@ -185,11 +185,14 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
...
     all_nodes.each do |n|
-      if neutron["attributes"]["neutron"]["networking_mode"] == "gre"
-        net_svc.allocate_ip "default", "os_sdn", "host", n
-      else
-        net_svc.enable_interface "default", "nova_fixed", n
-        if neutron["attributes"]["neutron"]["networking_mode"] == "vlan"
+      # TODO(toabctl): The same code is in the neutron barclamp. Should be extracted and reused!
+      #                (see crowbar_framework/app/models/neutron_service.rb)
+      if neutron["attributes"]["neutron"]["networking_plugin"] == "ml2"
+	if neutron["attributes"]["neutron"]["ml2_type_drivers"].include?("gre")
```
Comment (rhafer) [inline crowbar_framework/app/models/nova_service.rb]: Wrong indentation. Use 2 spaces.

File: crowbar_framework/app/models/nova_service.rb [hunk 2]
```diff
@@ -199,7 +202,7 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
             node.crowbar["crowbar"]["network"]["nova_fixed"]["use_vlan"] = false
             node.save
           end
-        end
+	end
```
Comment (rhafer) [inline crowbar_framework/app/models/nova_service.rb]: Same here.

**Comment by rhafer** [comment]
LGTM, apart form the style nit picks.

File: crowbar_framework/app/models/nova_service.rb (hunk 1 above)
Comment (toabctl) [inline crowbar_framework/app/models/nova_service.rb]: > Wrong indentation. Use 2 spaces.

Done in https://github.com/crowbar/barclamp-nova/commit/a1841c17f6a764b84e31838f1b4c06f28ee887b3

File: crowbar_framework/app/models/nova_service.rb (hunk 2 above)
Comment (toabctl) [inline crowbar_framework/app/models/nova_service.rb]: > Same here.

Done in https://github.com/crowbar/barclamp-nova/commit/a1841c17f6a764b84e31838f1b4c06f28ee887b3

**Comment by dirkmueller** [comment]
+1

**Comment by rhafer** [comment]
+1

**Pull request merged**